cmd/podman/machine/info.go

```diff
@@ -0,0 +1,166 @@
+	"gopkg.in/yaml.v2"
+)
+
+var infoDescription = `Display information pertaining to the machine host.`
+
+var (
+	infoCmd = &cobra.Command{
+		Use:               "info [options]",
+		Short:             "Display machine host info",
+		Long:              infoDescription,
+		RunE:              info,
+		Args:              validate.NoArgs,
+		ValidArgsFunction: completion.AutocompleteNone,
+		Example:           `podman machine info`,
+	}
+)
+
+var (
+	inFormat string
+)
+
+// Info contains info on the machine host and version info
+type Info struct {
+	Host    *HostInfo      `json:"Host"`
+	Version define.Version `json:"Version"`
+}
+
+// HostInfo contains info on the machine host
+type HostInfo struct {
+	Arch             string `json:"Arch"`
+	DefaultMachine   string `json:"DefaultMachine"`
+	EventsDir        string `json:"EventsDir"`
+	MachineImageDir  string `json:"MachineImageDir"`
+	MachineConfigDir string `json:"MachineConfigDir"`
+	NumberOfMachines int    `json:"NumberOfMachines"`
+	OS               string `json:"OS"`
+	RemoteSocket     string `json:"RemoteSocket"`
+	RunningMachine   string `json:"RunningMachine"`
+	VMType           string `json:"VMType"`
+}
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: infoCmd,
+		Parent:  machineCmd,
+	})
+
+	flags := infoCmd.Flags()
+	formatFlagName := "format"
+	flags.StringVarP(&inFormat, formatFlagName, "f", "", "Change the output format to JSON or a Go template")
+	_ = infoCmd.RegisterFlagCompletionFunc(formatFlagName, common.AutocompleteFormat(&define.Info{}))
+}
+
+func info(cmd *cobra.Command, args []string) error {
+	info := Info{}
+	version, err := define.GetVersion()
+	if err != nil {
+		return errors.Wrapf(err, "error getting version info")
+	}
+	info.Version = version
+
+	host, err := hostInfo()
+	if err != nil {
+		return err
+	}
+	info.Host = host
+
+	switch {
+	case report.IsJSON(inFormat):
+		b, err := json.MarshalIndent(info, "", "  ")
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(b))
+	case cmd.Flags().Changed("format"):
+		// Cannot use report.New() as it enforces {{range .}} for OriginUser templates
+		tmpl := template.New(cmd.Name()).Funcs(template.FuncMap(report.DefaultFuncs))
+		inFormat = report.NormalizeFormat(inFormat)
+		tmpl, err := tmpl.Parse(inFormat)
+		if err != nil {
+			return err
+		}
+		return tmpl.Execute(os.Stdout, info)
+	default:
+		b, err := yaml.Marshal(info)
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(b))
+	}
+
+	return nil
+}
+
+func hostInfo() (*HostInfo, error) {
+	host := HostInfo{}
+
+	host.Arch = runtime.GOARCH
+	host.OS = runtime.GOOS
+
+	provider := GetSystemDefaultProvider()
+	var listOpts machine.ListOptions
+	listResponse, err := provider.List(listOpts)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get machines")
+	}
+
+	host.NumberOfMachines = len(listResponse)
+	host.RemoteSocket = fmt.Sprintf("/run/user/%d/podman/podman.sock", os.Getuid())
```

Comment:
This is the only field I'm not sure is entirely correct, please correct me if I'm wrong here.